app/src/main/java/org/indywidualni/fblite/activity/MainActivity.java

```diff
@@ -723,6 +759,33 @@ protected void onNewIntent(Intent intent) {
         }
     }
 
+    private void showLongPressedImageMenu(ContextMenu menu, String imageUrl) {
+        mPendingImageUrlToSave = imageUrl;
+        menu.add(0, ID_CONTEXT_MENU_SAVE_IMAGE, 0, getString(R.string.save_img));
+    }
+
+    private void saveImageToDisk(String imageUrl) {
+        File imageStorageDir = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES), "FaceSlim");
+
+        if (!imageStorageDir.exists()) {
+            imageStorageDir.mkdirs();
+        }
+
+        SimpleDateFormat sdatum = new SimpleDateFormat("yyyyMMdd_HHmmss");
```

Comment:
`yyyyMMdd_HHmmss` instead of `System.currentTimeMillis()` because it's human readable and it's the same naming system as with default Android camera app.